### PR TITLE
Support search/open a trace by traceId in lens

### PR DESCRIPTION
**Pull request by tacigar**
Related: #2352

### CODE REVIEW

**Comment by tacigar** [comment]
I used GlobalSearch when implementing this feature.
The user can specify traceId in the same way as other search conditions.
If traceId is specified when the search button is pressed, jump to tracePage without searching even if other search conditions are specified.

File: zipkin-lens/src/components/GlobalSearch/ConditionAnnotationQuery.js [hunk 1]
```diff
@@ -12,8 +12,6 @@ const propTypes = {
   isFocused: PropTypes.bool.isRequired,
 };
 
-// TODO: After specific-tags feature is completed in server-side,
```
Comment (codefromthecrypt) [inline zipkin-lens/src/components/GlobalSearch/ConditionAnnotationQuery.js:15]: :D

Comment (tacigar) [reply zipkin-lens/src/components/GlobalSearch/ConditionAnnotationQuery.js:15]: :)

**Comment by tacigar** [comment]
traceid

**Comment by codefromthecrypt** [comment]
haven't installed locally yet. is there a way to have this work with hyperlink if not already? I mean a link with a trace ID embedded shortcut to the trace view screen?

**Comment by codefromthecrypt** [comment]
so navigating from the search bar is a go!

http://localhost:9411/zipkin/trace/5c50102aa5ef50d0ba078c79dcd66bef doesn't work because we haven't re-routed "zipkin/trace/{traceid}" back to the index page I think.. looking into it

**Comment by codefromthecrypt** [comment]
looks like the old route was like this: /zipkin/traces/{id} not /zipkin/trace/{id}

I think we should follow this for now because it makes lens drop-in replacement of the old one. Later we can consider what @llinder proposed earlier which was hash routing

**Comment by codefromthecrypt** [comment]
fyi https://github.com/openzipkin/zipkin/blob/master/zipkin-autoconfigure/ui/src/main/java/zipkin2/autoconfigure/ui/ZipkinUiAutoConfiguration.java#L134

**Comment by tacigar** [comment]
@adriancole 
OK, thank you!!
I'll change them in this PR now.

**Comment by tacigar** [comment]
Fixed

**Comment by tacigar** [comment]
> haven't installed locally yet. is there a way to have this work with hyperlink if not already? I mean a link with a trace ID embedded shortcut to the trace view screen?

Yes, there is.

You mean this link, right?
link

**Comment by codefromthecrypt** [comment]
this question was about people who use bookmarks. you already addressed
this with the /zipkin/traces change.

basically we want existing bookmarks to work at least for a while I think.

**Review by codefromthecrypt — APPROVED** [review]
thanks it works

File: zipkin-lens/src/components/GlobalSearch/ConditionTraceId.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+class ConditionTraceId extends React.Component {
+  constructor(props) {
+    super(props);
+    this.inputRef = undefined;
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+  }
+
+  handleKeyDown(e) {
+    if (e.keyCode === 13) {
```
Comment (zeagord) [inline zipkin-lens/src/components/GlobalSearch/ConditionTraceId.js]: KeyCode is deprecated. We must use the `e.key` instead. More better to follow the instruction here everywhere we handle keyboard events. 
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode#Example

**Review by zeagord — CHANGES_REQUESTED** [review]

File: zipkin-lens/src/components/GlobalSearch/ConditionTraceId.js (hunk 2 above)
Comment (zeagord) [reply zipkin-lens/src/components/GlobalSearch/ConditionTraceId.js]: LGTM other than that. Cheers.

Comment (tacigar) [reply zipkin-lens/src/components/GlobalSearch/ConditionTraceId.js]: Thank you for information 🙇 
I only use `event.key` for now.

**Comment by tacigar** [comment]
> this question was about people who use bookmarks. you already addressed
this with the /zipkin/traces change. basically we want existing bookmarks to work at least for a while I think.

Ah, I got it.
Thank you~

**Comment by codefromthecrypt** [comment]
good work folks

**Pull request merged**